### DISCUSSION TITLE
Make the lists collapsible (like in Roam)

### DISCUSSION THREAD

**Discussion by Stvad** [roam-garden/i#23] [Feature requests & Ideas] [answered]
Collapsible lists for the garden. That's important for legibility of deeply nested things. 

**Things to consider**:
- Preserve the state for user a in the [[local storage]]
- It can also be useful to have a default level at which the lists come in the "collapsed" state. Anything below level 3 can be a starting point [[to experiment]] from. 
- Theoretically this can be extracted from Roam Data, but it's not a part of JSON export atm

### Implementation considerations

**Workflowy Opensource clones to learn from**
- https://libs.garden/typescript/suyash/flowy
- https://github.com/adharris/react-workflowy-clone

**Libraries that may serve as a foundation**
- https://github.com/incepter/react-expansion

**Comment by Stvad** [comment, marked as answer]
progress https://twitter.com/VladyslavSitalo/status/1370249616885653506?s=20

**Reply by Stvad** [reply]
And it's live! https://twitter.com/VladyslavSitalo/status/1370825692133298176?s=20 
With example at https://vlad.roam.garden/. 

If you want to get this in your garden - you  need to go through the standard upload process at http://roam.garden and the new version will include the updates 
🙂

**Comment by mitchi-caminante** [comment]
Is there any way to make the collpased lists the default?

**Reply by Stvad** [reply]
Hey! not at the moment :(  
that information (about collapse state) is not included in the original export json. It should be possible to work around it if one uses the plugin #28 for publishing, but that's not implemented yet.  

Another approach can be to have special "control tags" as discussed in #19

**Reply by mitchi-caminante** [reply]
Thanks for the reply! It's not a huge issue for me, it's just my personal preference, I think it looks tidier and doesn't throw a ton of info at people right away 🙂
